yarn-project/aztec.js/src/contract/deploy_sent_tx.ts

```diff
@@ -24,9 +23,9 @@ export type DeployTxReceipt<TContract extends ContractBase = Contract> = FieldsO
  */
 export class DeploySentTx<TContract extends Contract = Contract> extends SentTx {
   constructor(
-    private artifact: ContractArtifact,
     wallet: PXE | Wallet,
     txHashPromise: Promise<TxHash>,
+    private postDeployCtor: (address: AztecAddress, wallet: Wallet) => Promise<Contract>,
```

Comment:
It broke a circular dependency between contract, sent_tx, and deploy_tx.
TBH I wasn't aware that this DeployMethod was being constructed publically, hence this has actually affected e.g. our browser test. But I'm also not sure it should ever need to be constructed by an end user so expect that's us just providing a bad api.

Reply:
> It broke a circular dependency between contract, sent_tx, and deploy_tx.

Don't we still have the circular dependency, since we are still importing `type Contract`? If so, and if the main reason for breaking circular deps is a cognitive and not a runtime performance one, then I'd argue that this change doesn't make code easier to follow (it's actually probably harder). Instead, it just pleases the linter, since it explicitly ignores type imports. But please disregard this comment if the circular dep was elsewhere and I missed it.

> I wasn't aware that this DeployMethod was being constructed publically, hence this has actually affected e.g. our browser test. But I'm also not sure it should ever need to be constructed by an end user so expect that's us just providing a bad api.

Agree, this class probably shouldn't be used directly by the end user.